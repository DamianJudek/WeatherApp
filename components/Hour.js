import React from 'react';
import { View, Text, StyleSheet, Image } from 'react-native';

const styles = StyleSheet.create({
  icon: {
    height: 60,
    width: 60,
  },
  baseText: {
    fontFamily: 'Roboto',
    color: '#fff',
    fontSize: 18,
  },
});

const Hour = (props) => {
  const { weather } = props;
  return (
    <View style={props.style}>
      <Text style={styles.baseText}>{weather.hour}</Text>
      <Image
        style={styles.icon}
        resizeMode="contain"
        source={{
          uri: `http://openweathermap.org/img/wn/${weather.icon}@2x.png`,
        }}
      />
      <Text style={styles.baseText}>{`${weather.temp}\u2103`}</Text>
    </View>
  );
};

export default Hour;
