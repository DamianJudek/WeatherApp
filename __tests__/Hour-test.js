import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Hour from '../components/Hour';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
test('renders correctly', () => {
  const snap = renderer
    .create(
      <Hour
        weather={{
          dayNumber: 6,
          description: 'clear sky',
          hour: '06:00',
          humidity: 88,
          icon: '03d',
          temp: 17,
          wind: 2.93,
        }}
        style={[
          {
            alignItems: 'center',
            borderColor: '#fff',
            borderRightWidth: 1,
            flex: 1,
            justifyContent: 'center',
          },
          { borderRightWidth: 0 },
        ]}
      />,
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});
