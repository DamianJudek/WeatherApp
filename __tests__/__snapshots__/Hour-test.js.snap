// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#fff",
      "borderRightWidth": 0,
      "flex": 1,
      "justifyContent": "center",
      "opacity": 0,
      "transform": Array [
        Object {
          "translateY": 35,
        },
      ],
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Roboto",
        "fontSize": 18,
      }
    }
  >
    06:00
  </Text>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "http://openweathermap.org/img/wn/03d@2x.png",
      }
    }
    style={
      Object {
        "height": 60,
        "width": 60,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Roboto",
        "fontSize": 18,
      }
    }
  >
    17â„ƒ
  </Text>
</View>
`;
